lib/code_corps/user_task_matcher.ex

```diff
@@ -0,0 +1,25 @@
+defmodule CodeCorps.UserTaskMatcher do
+  @moduledoc """
+  Find the top tasks most matching a User's skills
+  """
+
+  alias CodeCorps.{Repo, Task, User, TaskSkill}
+  import Ecto.Query
+
+  @spec match_user(User.t, number) :: [Task.t]
+  def match_user(%CodeCorps.User{} = user, tasks_count) do
+    query = from t in TaskSkill,
+      join: skill in assoc(t, :skill),
+      join: user_skill in assoc(skill, :user_skills),
+      join: user in assoc(user_skill, :user),
+      where: user.id == ^user.id,
+      group_by: t.task_id,
+      order_by: count(t.task_id),
+      limit: ^tasks_count,
+      select: t.task_id
```

Comment:
Have you been able to run an EXPLAIN on this query by chance to see how well it performs?